pkg/cloud/services/ec2/subnets.go

```diff
@@ -146,6 +131,51 @@ LoopExisting:
 	return nil
 }
 
+func (s *Service) getDefaultSubnets() (infrav1.Subnets, error) {
+	zones, err := s.getAvailableZones()
+	if err != nil {
+		return nil, err
+	}
+
+	//TODO: should this be configurable or are we ok with 3 AZs max for the default?
+	if len(zones) > 3 {
+		s.scope.V(2).Info("region has more than 3 availability zones, picking 3 at random", "region", s.scope.Region())
+		rand.Shuffle(len(zones), func(i, j int) {
+			zones[i], zones[j] = zones[j], zones[i]
+		})
+		zones = append(zones, zones[0], zones[1], zones[2])
+	}
```

Comment:
We should probably either expose this as an option, or sort it and pick the first N.

Reply:
The reason i didn't sort and pick the first n is that if you have `us-east-1` with 6 AZs then it will always pick a,b,c and there is an uneven distribution across the AZs in the region. I thought it might be better to distribute across all the AZs to help in some small way with resource consumption.

I could make the maximum number of AZs configurable via a option and set the default to 3. I don't think we should create subnets in every AZ for a region as thats not required for a HA solution, 3 would be a good sensible default.  What do you think?

Reply:
I've added options to expose the maximum number of AZs to use and also whether to select randomly or by alphabetical order if there is more AZs than the max to use.

Reply:
These are the 2 options i have added:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/26f2a95af8888bc2a936a322a5f15ee549517206/api/v1alpha3/types.go#L217